code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -1019,3 +1022,64 @@
 
 		to_chat(src, output)
 
+/mob/living/carbon/human
+	var/next_sonar_ping = 0
+
+/mob/living/carbon/human/proc/sonar_ping()
+	set name = "Psychic Ping"
+	set desc = "Allows you to listen in to psychic traces of organisms around you."
+	set category = "Abilities"
+
+	if(incapacitated())
+		to_chat(src, "<span class='warning'>You need to recover before you can use this ability.</span>")
+		return
+	if(world.time < next_sonar_ping)
+		to_chat(src, "<span class='warning'>You need another moment to focus.</span>")
+		return
+	next_sonar_ping += 10 SECONDS
+	var/heard_something = FALSE
+	to_chat(src, "<span class='notice'>You take a moment to tune into the local Nlom...</span>")
+	var/dirs = list()
+	for(var/mob/living/L in range(20))
+		var/turf/T = get_turf(L)
+		if(!T || L == src || L.stat == DEAD || L.isSynthetic() || L.is_diona() || isvaurca(L) || L.invisibility == INVISIBILITY_LEVEL_TWO)
+			continue
+		heard_something = TRUE
+		var/image/ping_image = image(icon = 'icons/effects/effects.dmi', icon_state = "sonar_ping", loc = src)
+		ping_image.plane = LIGHTING_LAYER+1
+		ping_image.layer = LIGHTING_LAYER+1
+		ping_image.pixel_x = (T.x - src.x) * WORLD_ICON_SIZE
+		ping_image.pixel_y = (T.y - src.y) * WORLD_ICON_SIZE
+		src << ping_image
+		addtimer(CALLBACK(GLOBAL_PROC, /proc/qdel, ping_image), 8)
+		var/direction = num2text(get_dir(src, L))
+		var/dist
+		if(text2num(direction))
+			switch(get_dist(src, L) / client.view)
+				if(0 to 0.2)
+					dist = "very close"
+				if(0.2 to 0.4)
+					dist = "close"
+				if(0.4 to 0.6)
+					dist = "a little ways away"
+				if(0.6 to 0.8)
+					dist = "farther away"
+				else
+					dist = "far away"
+		else
+			dist = "on top of you"
+		LAZYINITLIST(dirs[direction])
+		dirs[direction][dist] += 1
+	for(var/d in dirs)
+		var/feedback = list()
+		if(isnull(d)) // can be 0
+			continue
+		for(var/dst in dirs[d])
+			if(isnull(dst))
```

Comment:
again is this even necessary?

Reply:
It's also here to differentiate between null and zero, but I'm not sure a null can actually be introduced. Probably just paranoia on my part.